src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.model.User;
+import java.util.concurrent.atomic.AtomicLong;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static RegistrationServiceImpl registrationService;
+    private static AtomicLong userIdCounter;
+
+    @BeforeAll
+    static void createReference() {
+        registrationService = new RegistrationServiceImpl();
+        userIdCounter = new AtomicLong(1);
+    }
+
+    @BeforeEach
+    void resetStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userAlreadyExists_shouldThrowException() {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("validUser123");
+        user.setPassword("password123");
+        user.setAge(20);
+        Storage.people.add(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin("validUser123");
+        duplicateUser.setPassword("password123");
+        duplicateUser.setAge(20);
+
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(duplicateUser));
+    }
```

Comment:
This test case checks if the exception is thrown when a duplicate user is added, which is good. However, the checklist specifies that you should use low-level methods like `Storage.people.add()` to add users for testing instead of the `register` method. In this case, you are correctly using `Storage.people.add()` to add the initial user, but please ensure that the `register` method itself does not use `Storage.people.add()` internally, as this would be testing the method with itself.